src/utils.zig

```diff
@@ -1467,6 +1488,25 @@ pub const BinaryWriter = struct {
 		self.data.deinit();
 	}
 
+	pub fn writeVec(self: *BinaryWriter, T: type, value: T) void {
+		return switch(@typeInfo(T)) {
+			.vector => |typeInfo| {
+				for(0..typeInfo.len) |i| {
```

Comment:
Should be an inline for. The compiler can generate better code if the element index of the vector is comptime known.